### PR TITLE
Add revenue functionality

### LINKED ISSUE #26: [ENHANCEMENT] Add "Convert Revenue" option to each row in TCD's governance page
Add an option that has 🔁 sign after the amount. On clicking, user can choose to (1) stake the revenue or (2) withdraw the revenue.

## Motivation
Currently the only way for user to convert users pending revenue to voting power while still maintaining the stake it to click `withdraw` then `deposit`. This is non-obvious for users.

## Suggested Implementation
Add a button to each row in table. Upon clicked, a modal pops up for users to choose. After chosen, a saga action is dispatch to perform the corresponding action.

### PR DESCRIPTION
**Pull request by yeast72**
Fixes #26 
- Add max withdraw button
- Add revenue column
- Add ConvertRevenue modal
- Add Convert revenue to stake button
- Add withdraw revenue button
- Add  stake revenue type badge in transaction popup

### CODE REVIEW

**Comment by sorawit** [comment]
@yeast72  Screenshots inside the pull request would be nice.  Also please mention that this fixes #26

File: app/src/sagas/tcd.js [hunk 1]
```diff
@@ -84,6 +89,15 @@ function* handleLoadTcds({ user, tokenAddress }) {
                   .div(new BN(totalOwnership))
               }
 
+              if (voterToTokenLock[user]) {
+                revenue = userStake.sub(new BN(voterToTokenLock[user]))
+                //  < -0.01 = BN(0)
+                if (revenue.lt(new BN(-0.01))) {
+                  revenue = new BN(0)
+                } else if (revenue.lt(new BN(0)) && revenue.gte(-0.01)) {
+                  throw new Error('Revenue greater than 0.01')
```
Comment (evilpeach) [inline app/src/sagas/tcd.js]: @yeast72 It should be `throw new Error('Revenue greater than -0.01')`

File: app/src/components/ConvertRevenueModal/ConvertRevenueModal.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+const mapStateToProps = (
+  state,
+  {
+    userRevenue,
+    tokenAddress,
+    dataSourceAddress,
+    tcdAddress,
+    userOwnership,
```
Comment (evilpeach) [inline app/src/components/ConvertRevenueModal/ConvertRevenueModal.js]: remove

File: app/src/components/ConvertRevenueModal/ConvertRevenueModal.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    totalOwnership,
+  },
+) => {
+  const community = communityDetailSelector(state, {
+    address: tokenAddress,
+  })
+  if (!community) return {}
+  return {
+    userRevenue,
+    userOwnership,
```
Comment (evilpeach) [inline app/src/components/ConvertRevenueModal/ConvertRevenueModal.js]: remove

**Pull request merged**